consensus/scp/tests/mock_network/mod.rs

```diff
@@ -563,17 +512,206 @@ impl SCPNode {
+                    "( testing ) externalized {}/{} values at node {}",
+                    cur_num_values,
+                    values.len(),
+                    node_id
+                );
+                break;
+            }
+
+            if prev_num_values != cur_num_values {
+                assert!(cur_num_values > prev_num_values);
+                prev_num_values = cur_num_values;
+            }
+
+            if last_log.elapsed().as_secs() > 1 {
+                log::info!(
+                    network.logger,
+                    "( testing ) externalized {}/{} values at node {}",
+                    cur_num_values,
+                    values.len(),
+                    node_id
+                );
+                last_log = Instant::now();
+            }
+        }
+
+        let all_values_are_correct = {
+            values.iter().cloned().collect::<HashSet<String>>()
+                == network
+                    .get_shared_data(&node_id)
+                    .get_all_values()
+                    .iter()
+                    .cloned()
+                    .collect::<HashSet<String>>()
+        };
+
+        if !all_values_are_correct {
+            log::error!(
+                network.logger,
+                "( testing ) node {} externalized wrong values!",
+                node_id
+            );
+            assert!(all_values_are_correct); // exit
+        }
+    }
+
+    // Check all blocks in the ledger are the same
+    let node0_data = network.get_shared_data(&test_utils::test_node_id(0)).ledger;
+
+    if !(node0_data.len() > 0) {
+        log::error!(
+            network.logger,
+            "failing 'node0_data.len() > 0' in run_test()"
+        );
+    }
+    assert!(node0_data.len() > 0);
+
+    for node_num in 0..num_nodes {
+        let node_data = network
+            .get_shared_data(&test_utils::test_node_id(node_num as u32))
+            .ledger;
+
+        if node0_data.len() != node_data.len() {
+            log::error!(
+                network.logger,
+                "failing 'node0_data.len() == node_data.len()' in run_test()"
+            );
+        }
+        assert_eq!(node0_data.len(), node_data.len());
+
+        for block_num in 0..node0_data.len() {
+            if node0_data.get(block_num) != node_data.get(block_num) {
+                log::error!(
+                    network.logger,
+                    "failing 'node0_data.get(block_num) == node_data.get(block_num)' in run_test()"
+                );
+            }
+            assert_eq!(node0_data.get(block_num), node_data.get(block_num));
+        }
+    }
 
-    output
+    log::info!(
+        logger,
+        "RESULTS,{},{},{},{},{},{}",
+        network_name,
+        start.elapsed().as_millis(),
+        values.len(),
+        options.submissions_per_sec,
+        options.max_values_per_slot,
+        options.scp_timebase.as_millis(),
+    );
+
+    // stop the threads
+    network.stop_all();
```

Comment:
This shouldn't be necessary, it should happen automatically when network goes out of scope.

Reply:
this is probably the source of complaint with Drop